prow/gerrit/adapter/adapter.go

```diff
@@ -358,6 +360,25 @@ func failedJobs(account int, revision int, messages ...gerrit.ChangeMessageInfo)
 	return failures
 }
 
+func (c *Controller) handleInRepoConfigError(err error, instance string, change gerrit.ChangeInfo) error {
+	if err != nil {
+		// If we have not already recorded this failure send an error essage
+		if failed, ok := c.inRepoConfigFailures[fmt.Sprintf("%s%s%s", instance, change.ID, change.CurrentRevision)]; !ok || !failed {
+			if setReviewWerr := c.gc.SetReview(instance, change.ID, change.CurrentRevision, inRepoConfigFailed, nil); setReviewWerr != nil {
```

Comment:
The `inRepoConfigFailures` map will not persist across restarts. This could cause us to leave an extra comment for every restart

Reply:
I considered this. It will usually be one a week. Even if we get to the point where we are auto-deploying once a day I don't think that a message every 24h is horrible. The alternative would be keeping this in lastSyncUpdate in gcs which feels like overkill.

When we do the larger refactor of the gerrit adapter to keep track of changes it needs to retry that will solve this problem.